generator/src/lib.rs

```diff
@@ -2464,15 +2468,137 @@ pub fn write_source_code(path: &Path) {
         #bitflags_macro
         #handle_nondispatchable_macro
         #define_handle_macro
+        #ptr_chain_code
+    };
+
+    let dir = vk_rs.as_ref().parent().unwrap().to_path_buf().join("vk");
```

Comment:
It would be simpler to just pass the root dir `ash/src` instead of `ash/src/vk.rs`. Then you don't need to call `parent`.